### PR TITLE
Add documentation for using gRPC on macOS

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/13544

Internal review URLs:
* https://review.docs.microsoft.com/en-us/aspnet/core/grpc/aspnetcore?view=aspnetcore-3.0&branch=pr-en-us-13568&tabs=visual-studio#grpc-and-aspnet-core-on-macos
* https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?view=aspnetcore-3.0&branch=pr-en-us-13568&tabs=visual-studio#run-the-service

### CODE REVIEW

File: aspnetcore/grpc/aspnetcore.md [hunk 1]
```diff
@@ -82,6 +82,45 @@ The gRPC API provides access to some HTTP/2 message data, such as the method, ho
 
 [!code-csharp[](~/grpc/aspnetcore/sample/GrcpService/GreeterService2.cs?highlight=6-7&name=snippet)]
 
+## gRPC and ASP.NET Core on macOS
+
+Kestrel doesn't support HTTP/2 with [Transport Layer Security (TLS)](https://tools.ietf.org/html/rfc5246) on macOS. The ASP.NET Core gRPC template and samples use TLS by default, and you will see this error message when you attempt to start the gRPC server:
```
Comment (scottaddie) [inline aspnetcore/grpc/aspnetcore.md]:
```suggestion
Kestrel doesn't support HTTP/2 with [Transport Layer Security (TLS)](https://tools.ietf.org/html/rfc5246) on macOS. The ASP.NET Core gRPC template and samples use TLS by default. You'll see the following error message when you attempt to start the gRPC server:
```

File: aspnetcore/grpc/aspnetcore.md [hunk 2]
```diff
@@ -82,6 +82,45 @@ The gRPC API provides access to some HTTP/2 message data, such as the method, ho
...
+
+> Unable to bind to https://localhost:5001 on the IPv4 loopback interface: 'HTTP/2 over TLS is not supported on OSX due to missing ALPN support.'.
+
+To workaround this issue you must configure Kestrel and the gRPC client to use HTTP/2 **without** TLS. You should only do this during development. Not using TLS will result in gRPC messages being sent without encryption.
```
Comment (scottaddie) [inline aspnetcore/grpc/aspnetcore.md]:
```suggestion
To work around this issue, configure Kestrel and the gRPC client to use HTTP/2 **without** TLS. You should only do this during development. Not using TLS will result in gRPC messages being sent without encryption.
```

File: aspnetcore/grpc/aspnetcore.md [hunk 3]
```diff
@@ -82,6 +82,45 @@ The gRPC API provides access to some HTTP/2 message data, such as the method, ho
...
+
+Kestrel must configure a HTTP/2 endpoint without TLS in `Program.cs`:
+
+```cs
```
Comment (scottaddie) [inline aspnetcore/grpc/aspnetcore.md]:
```suggestion
```csharp
```

File: aspnetcore/grpc/aspnetcore.md [hunk 4]
```diff
@@ -82,6 +82,45 @@ The gRPC API provides access to some HTTP/2 message data, such as the method, ho
...
+            webBuilder.ConfigureKestrel(options =>
+            {
+                // Setup a HTTP/2 endpoint without TLS.
+                options.ListenLocalhost(5000, o => o.Protocols = HttpProtocols.Http2);
+            });
+            webBuilder.UseStartup<Startup>();
+        });
+```
+
+The gRPC client must set the `System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport` switch to true and use `http` in the server address:
```
Comment (scottaddie) [inline aspnetcore/grpc/aspnetcore.md]:
```suggestion
The gRPC client must set the `System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport` switch to `true` and use `http` in the server address:
```

File: aspnetcore/grpc/aspnetcore.md [hunk 5]
```diff
@@ -82,6 +82,45 @@ The gRPC API provides access to some HTTP/2 message data, such as the method, ho
...
+AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
+
+var httpClient = new HttpClient();
+// The port number(5000) must match the port of the gRPC server.
+httpClient.BaseAddress = new Uri("http://localhost:5000");
+var client = GrpcClient.Create<Greeter.GreeterClient>(httpClient);
+```
+
+> [!WARNING]
+> HTTP/2 without [Transport Layer Security (TLS)](https://tools.ietf.org/html/rfc5246) should only be used during app development. Production applications should always use transport security. Visit [Security considerations in gRPC for ASP.NET Core](xref:grpc/security#transport-security) for more information.
```
Comment (scottaddie) [inline aspnetcore/grpc/aspnetcore.md]:
```suggestion
> HTTP/2 without [Transport Layer Security (TLS)](https://tools.ietf.org/html/rfc5246) should only be used during app development. Production applications should always use transport security. For more information, see [Security considerations in gRPC for ASP.NET Core](xref:grpc/security#transport-security).
```

File: aspnetcore/tutorials/grpc/grpc-start.md [hunk 6]
```diff
@@ -118,6 +118,9 @@ info: Microsoft.Hosting.Lifetime[0]
 info: Microsoft.Hosting.Lifetime[0]
 ```
 
+> [!NOTE]
+> Additional configuration is required to successfully run gRPC services with ASP.NET Core on macOS. Visit [gRPC and ASP.NET Core on macOS](xref:grpc/aspnetcore#grpc-and-aspnet-core-on-macos) for more information.
```
Comment (scottaddie) [inline aspnetcore/tutorials/grpc/grpc-start.md]:
```suggestion
> Additional configuration is required to successfully run gRPC services with ASP.NET Core on macOS. For more information, see [gRPC and ASP.NET Core on macOS](xref:grpc/aspnetcore#grpc-and-aspnet-core-on-macos).
```

**Comment by JamesNK** [comment]
All feedback applied. This is ready to merge. I'd like this to be deployed soon (today/tomorrow) because there is a new link to it in the preview 8 gRPC template.

**Comment by JamesNK** [comment]
// @Rick-Anderson @guardrex

**Comment by guardrex** [comment]
The original issue was marked for Pre9. This works today? If this is for Pre8, then it would go on the staging branch (`release/3.0-preview8`) and be merged when Pre8 is released.

**Comment by JamesNK** [comment]
Yes this works today. I would like it merged sometime soon so that the link in the upcoming preview 8 template can be tested.

**Pull request merged**